lib/dgram.js

```diff
@@ -439,9 +578,11 @@ Socket.prototype.send = function(buffer,
                                    ['Buffer', 'string'], buffer);
   }
 
-  port = port >>> 0;
-  if (port === 0 || port > 65535)
-    throw new ERR_SOCKET_BAD_PORT(port);
+  if (notConnected) {
+    port = port >>> 0;
```

Comment:
Not really related to this PR, but I wonder if using `isLegalPort()` is an option here. If you want, you can try it in a separate commit here, or I can try in a follow up PR.